File: eAnimalShelter/eAnimalShelter.Model/SearchObjects/DonationSearchObject.cs
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Model.SearchObjects
{
    public class DonationSearchObject : BaseSearchObject
    {
        public string? UserFullName { get; set; }
        public int? UserId { get; set; }

        public DonationStatus? TransactionStatus { get; set; }

        public DateTime? FromDate { get; set; }

        public DateTime? ToDate { get; set; }

        public decimal? MinAmount { get; set; }

        public decimal? MaxAmount { get; set; }
    }
}

File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/search_objects/donation_search_object.dart
class DonationSearchObject {
  String? userFullName;

  double? minAmount;
  double? maxAmount;

  int? transactionStatus;

  String? sortBy;

  int page;
  int pageSize;

  DonationSearchObject({
    this.userFullName,
    this.minAmount,
    this.maxAmount,
    this.transactionStatus,
    this.sortBy,
    this.page = 1,
    this.pageSize = 10,
  });

  Map<String, dynamic> toJson() {
    return {
      "userFullName": userFullName,
      "minAmount": minAmount,
      "maxAmount": maxAmount,
      "transactionStatus":transactionStatus,
      "sortBy": sortBy,
      "page": page,
      "pageSize": pageSize,
      "includeTotalCount": true,
    };
  }
}